module_utils/network_lsr/utils.py

```diff
@@ -126,6 +126,29 @@ def GMainLoop_iterate_all(cls):
     def create_cancellable(cls):
         return cls.Gio().Cancellable.new()
 
+    @classmethod
+    def create_callback(cls, finish_method):
+        """
+        Create a callback that will return the result of the finish method and
+        quit the GMainLoop
+
+        :param finish_method str: Name of the finish method to call from the
+        source object in the callback
+        """
+
+        def callback(source_object, res, user_data):
+            success = None
+            try:
+                success = getattr(source_object, finish_method)(res)
```

Comment:
Maybe resolve the closure first, and not during the callback? Then, on invalid function name, the exception happens synchronously during `create_callback()`. Also, I feels slightly nicer to resolve the function name at the moment when creating the callback, then later.

Reply:
Yes, it would be better but I do not see that it is possible since `source_object` is not defined in `create_callback()`. Let me know if I am missing something.

Reply:
you are right of course. And having it available would make the function more cumbersome to use. Good then!